rtloader/rtloader/api.cpp

```diff
@@ -357,39 +352,85 @@ void signalHandler(int sig, siginfo_t *, void *)
     }
 #    endif
 
-    // dump core if so configured
-    __sync_synchronize();
-    if (core_dump) {
-        core_dump(sig);
+    // Chain to the previous signal handler (typically Go runtime's handler)
+    if (old_sigsegv_handler.sa_flags & SA_SIGINFO) {
+        // Old handler uses the three-argument form
+        if (old_sigsegv_handler.sa_sigaction != NULL) {
+            old_sigsegv_handler.sa_sigaction(sig, info, context);
+        }
     } else {
-        kill(getpid(), SIGABRT);
+        // Old handler uses the simple one-argument form
+        if (old_sigsegv_handler.sa_handler != SIG_DFL && old_sigsegv_handler.sa_handler != SIG_IGN) {
+            old_sigsegv_handler.sa_handler(sig);
+        } else {
+            // No previous handler or it was default/ignore, so just abort
+            std::cerr << "Received SIGSEGV and no handler to chain to. Aborting. \n";
+            kill(getpid(), SIGABRT);
+        }
     }
 }
 
 /*
  * C-land crash handling
  */
-DATADOG_AGENT_RTLOADER_API int handle_crashes(const int enable, char **error)
+DATADOG_AGENT_RTLOADER_API int handle_crashes(const int enable_coredump, const int enable_stacktrace, char **error)
```

Comment:
Should we try to handle coredump and stacktraces independently when one errors ? Ie. not fail immediately if enable_coredump fails but still try to enable stacktrace too

Reply:
That's a good idea. The problem is that if enable_coredump fails and we carry on, we lose the error context for the failure. That will make it harder to debug.